docs/guides/operator/installation.adoc

```diff
@@ -21,6 +21,66 @@ The recommended way to install the {project_name} Operator in Kubernetes environ
 
 * Be sure that you have cluster-admin permission or an equivalent level of permissions granted by an administrator.
 
+[WARNING]
+====
+*Important: Automatic OLM Upgrades*
+
+By default, OLM automatically updates the {project_name} Operator when a new version is released. This can cause several significant issues:
+
+* When using the default {project_name} image, the Operator by default uses a matching image of the corresponding {project_name} version, resulting in *unintended {project_name} upgrades* when the Operator is upgraded.
+
+* Even when using custom images, major Operator upgrades can introduce compatibility issues with your existing Keycloak CR configuration.
```

Comment:
Ideally it's only on major releases. There could be a word of caution on minor releases as well because of operator logic changes.